aspnetcore/signalr/messagepackhubprotocol.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Use MessagePack Hub Protocol in SignalR for ASP.NET Core
+author: rachelappel
+description: Add MessagePack Hub Protocol to ASP.NET Core SignalR.
+manager: wpickett
+monikerRange: '>= aspnetcore-2.1'
+ms.author: rachelap
+ms.custom: mvc
+ms.date: 06/04/2018
+ms.prod: aspnet-core
+ms.technology: aspnet
+ms.topic: article
+uid: signalr/messagepackhubprotocol
+---
+
+# Use MessagePack Hub Protocol in SignalR for ASP.NET Core
+
+By [Brennan Conroy](https://github.com/BrennanConroy)
+
+This article assumes the reader is familiar with the topics covered in [Get Started](xref:signalr/get-started).
+
+## What is MessagePack?
+
+[MessagePack](https://msgpack.org/index.html) is a binary serialization format that is fast and compact. It's useful when performance and bandwidth are a concern because it creates smaller packets compared to [JSON](https://www.json.org/). Because it's a binary format, messages are unreadable when looking at network traces and logs unless the bytes are passed through a MessagePack parser. SignalR has added built-in support for the MessagePack format, and provides APIs for the client and server to use.
+
+## Configure MessagePack on the server
+
+To enable the MessagePack Hub Protocol on the server, install the `Microsoft.AspNetCore.SignalR.Protocols.MessagePack` package in your app. Add `AddMessagePackProtocol` to the `AddSignalR` call to enable MessagePack support on the server.
+
+> [!NOTE]
+> Json is enabled by default. Adding MessagePack enables support for both JSON and MessagePack clients.
+
+```csharp
+services.AddSignalR()
+    .AddMessagePackProtocol();
+```
+
+To customize how MessagePack will format your data, `AddMessagePackProtocol` takes a Func for configuring options. A property named `FormatterResolvers` is exposed in the options. The `FormatterResolvers` is a list of resolvers that are used by MessagePack to determine the format messages will be in. For more information on how the resolvers work, visit the MessagePack library at [MessagePack-CSharp](https://github.com/neuecc/MessagePack-CSharp). Attributes can be used on the objects you want to serialize to define how they should be handled.
```

Comment:
takes a ~~Func~~ **delegate** for configuring options

Reply:
~~A property named `FormatterResolvers` is exposed in the options. The `FormatterResolvers` is a list of resolvers that are used by MessagePack to determine the format messages will be in.~~

In that delegate, the `FormatterResolvers` property can be used to configure MessagePack serialization options. For more information on how the resolvers work, ...